src/basic/random-util.c

```diff
@@ -116,6 +117,15 @@ int rdrand(unsigned long *ret) {
 #endif
 
                 have_rdrand = !!(ecx & bit_RDRND);
+
+                if (have_rdrand > 0) {
+                        /* Allow disabling use of RDRAND with SYSTEMD_RDRAND=0
+                           If it is unset getenv_bool_secure will return a negative value. */
+                        if (!getenv_bool_secure("SYSTEMD_RDRAND")) {
```

Comment:
This is almost OK, but please make it `getenv_bool_secure("SYSTEMD_RDRAND") == 0`. The reason is that we avoid operations with `!` on integers. `getenv_bool_secure()` returns negative on error, which we want to treat the same as SYSTEMD_RDRAND=1, so the expression you used would have the same effect, but is misleading.

Reply:
Addressed.